public/app/plugins/datasource/prometheus/querycache/QueryCache.ts

```diff
@@ -62,26 +67,112 @@ export function getTargSig(targExpr: string, request: DataQueryRequest<PromQuery
  * Ident: Identity: the string that is not expected to change
  * Sig: Signature: the string that is expected to change, upon which we wipe the cache fields
  */
-type Milliseconds = number;
 export class QueryCache {
-  private overlapWindow: Milliseconds;
+  private overlapWindowMs;
+  private perfObeserver?: PerformanceObserver;
+  private shouldProfile: boolean;
+
+  pendingRequestIdsToTargSigs = new Map<
+    RequestID,
```

Comment:
are requestIds unique for each instantiation, or will they repeat at some point?

Reply:
they are unique to each time the PanelQuery runner is called